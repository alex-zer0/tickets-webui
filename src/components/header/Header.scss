.Header {
    &__image {
        width: 100%;
    }

    &__title {
        font-size: 1rem;
        padding: 1rem 3rem;
        box-shadow: 0 2px 2px rgba(0,0,0,.1);
    }
}
