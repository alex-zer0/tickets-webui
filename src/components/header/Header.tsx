import React from 'react';
import './Header.scss';

export interface HeaderProps {
  title: string
  imageUrl: string
  pageTitle?: string
}

export default function Header(props: HeaderProps) {
  const { title, imageUrl, pageTitle } = props;
  return (
    <header className='Header'>
      <img className='Header__image' src={imageUrl} alt={name} />
      <h1 className='Header__title'>
        {title}
        {pageTitle && ' > ' + pageTitle}
      </h1>
    </header>
  );
}
