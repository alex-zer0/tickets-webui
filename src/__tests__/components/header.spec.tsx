import React from 'react';
import renderer from 'react-test-renderer';
import Header, { HeaderProps } from '../../components/header/Header';

describe('COMPONENTS: Header', () => {
  it('renders w/o exceptions', () => {
    const props: HeaderProps = {
      title: 'Event name',
      imageUrl: 'http://banner-image-url',
    };

    const tree = renderer
      .create(<Header {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
