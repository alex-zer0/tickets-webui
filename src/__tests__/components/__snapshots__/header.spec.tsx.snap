// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS: Header renders w/o exceptions 1`] = `
<header
  className="Header"
>
  <img
    alt="nodejs"
    className="Header__image"
    src="http://banner-image-url"
  />
  <h1
    className="Header__title"
  >
    Event name
  </h1>
</header>
`;
